cli/src/commands/git/colocation.rs

```diff
@@ -0,0 +1,343 @@
+        GitColocationCommand::Disable(args) => cmd_git_colocation_disable(ui, command, args),
+        GitColocationCommand::Status(args) => cmd_git_colocation_status(ui, command, args),
+    }
+}
+
+fn cmd_git_colocation_status(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    _args: &GitColocationStatusArgs,
+) -> Result<(), CommandError> {
+    let workspace_command = command.workspace_helper(ui)?;
+    let is_colocated =
+        is_colocated_git_workspace(workspace_command.workspace(), workspace_command.repo());
+
+    if is_colocated {
+        writeln!(ui.stdout(), "Repository is currently colocated with Git.")?;
+        writeln!(
+            ui.hint_default(),
+            "To disable colocation, run: `jj git colocation disable`"
+        )?;
+    } else {
+        writeln!(
+            ui.stdout(),
+            "Repository is currently not colocated with Git."
+        )?;
+        writeln!(
+            ui.hint_default(),
+            "To enable colocation, run: `jj git colocation enable`"
+        )?;
+    }
+
+    Ok(())
+}
+
+fn cmd_git_colocation_enable(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    _args: &GitColocationEnableArgs,
+) -> Result<(), CommandError> {
+    let workspace_command = command.workspace_helper(ui)?;
+
+    // Check if the repo is already colocated before proceeding
+    if is_colocated_git_workspace(workspace_command.workspace(), workspace_command.repo()) {
+        writeln!(ui.status(), "Repository is already colocated with Git.")?;
+        return Ok(());
+    }
+
+    let workspace_root = workspace_command.workspace_root();
+    let dot_jj_path = workspace_root.join(".jj");
+    let jj_repo_path = dot_jj_path.join("repo");
+    let git_store_path = jj_repo_path.join("store").join("git");
+    let git_target_path = jj_repo_path.join("store").join("git_target");
+    let dot_git_path = workspace_root.join(".git");
+
+    // This is called after checking that the repo is backed by git, is not
+    // already colocated, and is a main workspace, but we must still bail out
+    // if a git repo already exist at the root folder
+    if dot_git_path.exists() {
+        return Err(user_error(
+            "A .git directory already exists in the workspace root. Cannot colocate.",
+        ));
+    }
+
+    // Update the git_target file to point to the new location of the git repo
+    // Note that we do this first so that it is easier to revert the operation
+    // in case there is a failure in this step or the next
+    let git_target_content = "../../../.git";
+    std::fs::write(&git_target_path, git_target_content).context(git_target_content)?;
+
+    // Move the git repository from .jj/repo/store/git to .git
+    if let Err(e) = std::fs::rename(&git_store_path, &dot_git_path) {
+        // Attempt to delete git_target_path if move fails and show an error message
+        let _ = std::fs::remove_file(&git_target_path);
```

Comment:
Better to leave files on unexpected error. We can do rename first and update the pointer file. It's relatively easy for user to recover from rename error than restoring the original `git_target` file.